test/utilities.js

```diff
@@ -232,6 +232,42 @@ describe('utilities', function () {
     expect(expect('foo').result()).to.equal('result');
   });
 
+  it('addMethod returns new assertion with flags copied over', function () {
+    var assertionConstructor;
+
+    chai.use(function(_chai, utils) {
+      assertionConstructor = _chai.Assertion;
+      _chai.Assertion.addMethod('returnNewAssertion', function () {
+        utils.flag(this, 'mySpecificFlag', 'value1');
+        utils.flag(this, 'ultraSpecificFlag', 'value2');
+      });
+    });
+
+    assertion1 = expect('foo');
+    assertion2 = assertion1.to.returnNewAssertion();
+
+    // Checking if a new assertion was returned
+    expect(assertion1).to.not.be.equal(assertion2);
+
+    // Checking if it's really an instance of an Assertion
+    expect(assertion2).to.be.instanceOf(assertionConstructor);
+
+    // If it is, calling length on it should return an assertion, not a function
+    var anAssertion = expect([1, 2, 3]).to.be.an.instanceof(Array);
+    expect(anAssertion.length).to.be.an.instanceof(assertionConstructor);
```

Comment:
Ah, that makes sense.
Thanks for the insight here.

I was also doing some research on [this](https://github.com/chaijs/chai/blob/master/lib/chai/utils/addChainableMethod.js#L85) too, shouldn't we create another assertion and transfer flags to it instead of returning `this`?

I'm not sure this is needed, I will do some further experiments with this, but it may be good to prevent further errors related to this.